lib/engine/game/g_1840/step/buy_sell_par_shares.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_sell_par_shares'
+
+module Engine
+  module Game
+    module G1840
+      module Step
+        class BuySellParShares < Engine::Step::BuySellParShares
+          def actions(entity)
+            return [] unless entity == current_entity
+
+            actions = []
+            actions << 'buy_shares' if can_buy_any?(entity)
+            actions << 'par' if can_ipo_any?(entity)
+            actions << 'sell_shares' if can_sell_any?(entity)
+
+            actions << 'pass' unless actions.empty? || !@game.all_major_corporations_ipoed?
```

Comment:
don't use unless with multiple conditions

Reply:
hehe, yeah that is borderline readable.